index.js

```diff
@@ -11,14 +12,40 @@ const nextButton = document.querySelector('[data-js="button-next"]');
 const pagination = document.querySelector('[data-js="pagination"]');
 
 // States
-const maxPage = 1;
-const page = 1;
+const maxPage = 42;
+let page = 1;
 const searchQuery = "";
 
+fetchCharacters("character", page);
+
+// Pagination
+
+nextButton.addEventListener("click", async () => {
+  page++;
+  if (page === 43) {
+    pagination.textContent = `1 / ${maxPage}`;
+  } else pagination.textContent = `${page} / ${maxPage}`;
+  if (page > 42) {
+    page = 1;
+  }
+  const data = await fetchCharacters("character", page);
+});
```

Comment:
The logic you need behind your nextButton is basically the following:
Always add 1 to `page` as long as page is smaller than `maxPage`.
In your `fetchCharacters` function you can then render the current values of `page` and `maxPage` in your pagination span.

And why don't you just call `fetchCharacters` as it is? Why do you save the return value in `data`? You do not do anything with that afterwards 🤔 
Additionally, you do not need an async function here.